templates/theme-default/layouts/partials/components/c-media.html

```diff
@@ -14,27 +14,58 @@
 {{- $media_desktop := or (index $params  (print "media_" $desktop_ratio)) $media -}}
 
 {{- $different_ratios := (ne $mobile_ratio $desktop_ratio) -}}
+{{- $only_img := and (hasPrefix ($media_mobile.mime_type | default "") "image") (hasPrefix ($media_desktop.mime_type | default "") "image") -}}
 
 {{- with $media -}}
   <figure class="c-media{{ with $focus_area }} c-media-{{ . }}{{ end }}{{ range $modifier }} c-media--{{ . }}{{ end }}{{ with $class_name }} {{ . }}{{end}}">
-    {{- partial "utils/asset" (dict
-      "globals" $globals
-      "context" $media_mobile
-      "options" (merge $params (dict
-        "class_name" (print "c-media__asset" (cond $different_ratios " md:u-hidden" "") )
-        "ratio" $mobile_ratio
-      ))
-    ) -}}
-    {{- if $different_ratios -}}
+    {{- if or (not $only_img) (not $different_ratios) -}}
       {{- partial "utils/asset" (dict
         "globals" $globals
-        "context" $media_desktop
+        "context" $media_mobile
         "options" (merge $params (dict
-          "class_name" (print "c-media__asset" (cond $different_ratios " -md:u-hidden" "") )
-          "ratio" $desktop_ratio
+          "class_name" (print "c-media__asset" (cond $different_ratios " md:u-hidden" "") )
+          "ratio" $mobile_ratio
+          "media" (cond $different_ratios "(max-width: 767px)" "")
```

Comment:
The `media` is still missing in the `c-image`. There preloads are set without media.